doc/architecture/decisions/0015-container-api.md

```diff
@@ -0,0 +1,103 @@
+# 15. Container API
+
+Date: 2018-09-17
+
+## Status
+
+Proposed
+
+## Context
+
+* We have several different roles for using a Holochain app (through a *container*):
+  * local, QML based UI components
+  * as a special case of the above: administration UI
+  * externally located (living in a browser) web UIs connected through some form of IPC
+  * *bridged* apps, i.e. Holochain app as the user of another Holochain app
```

Comment:
Is this suggesting that bridging communications should flow over a websocket channel? Feels like this should be direct function calls. Sorry, I may be conflating the other PR, this may indeed be just proposing a programming interface.

Reply:
No, this API is meant as the functions any network service project could use in order to implement a websocket channel, but that would be *setting up a user that is bound to that websocket session* and not in the scope of a container.

The container just provides the means to spawn Holochain instances and make their capabilities available. Bridging would be a use-case of that where that happens within the container, and definitely within the process the container runs in.

it's just that from the perspective of the container (and Holochain apps fwiw) we need the same semantics for all these (quite different) cases.